homeworks/3/ChatApp/src/com/company/Server.java

```diff
@@ -0,0 +1,148 @@
+package com.company;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class Server implements Runnable {
+
+    private ArrayList<ConnectionHandler> connectionHandlers;
+    private ServerSocket server;
+    private boolean done;
+    private ExecutorService pool;
+    private int port = 9999;
+
+    public Server() {
+        connectionHandlers = new ArrayList<>();
+        done = false;

```

Comment:
This variable can be removed because the only way its value gets changed is with the `shutdown` method which is called on line 45 after `IOException` has been caught. That happens after the only place where you use the variable (line 37).